setup.py

```diff
@@ -30,7 +30,7 @@
 
 EXTRAS_REQUIRE = {
     "neural-compressor": "neural-compressor",
-    "openvino": "openvino>=2022.2.0",
+    "openvino": ["openvino>=2022.2.0", "nncf"],
```

Comment:
Can we have nncf as an additional extra, instead of included in OpenVINO? I would like to keep the option of only using OpenVINO Runtime, with as few as possible dependencies. This is essential for using OpenVINO for deployment.
You would then do `pip install optimum-intel[openvino,nncf]` to install both OpenVINO and NNCF with optimum-intel.

Reply:
nncf is really a small dependency compared to transformers which are required as far as I understand. @helena-intel, do you mean that we can have only "openvino" as a dependency once we import the model from Hub? or we need optimum and transformers anyway?

Reply:
nncf is small, but it has dependencies which still add up. Dependencies can also have version conflicts with dependencies of other tools you need, so it is very useful to have an option to install optimum-intel[openvino] with the minimum number of dependencies. We discussed in the OpenVINO Runtime integration that in the future we could also remove PyTorch from the base install requirements, to make the footprint smaller.

These are current nncf dependencies (on Python 3.8/Linux): `addict, jsonschema, jstyleson, matplotlib, natsort, networkx, ninja, pandas, pillow, pydot, pymoo, pyparsing, scikit-learn, scipy, texttable, tqdm, wheel`

Reply:
From what I see now in the mandatory requirements for `pip install optimum[openvino]` we have:
```
INSTALL_REQUIRES = [
    "optimum",
    "transformers>=4.17.0",
    "datasets",
    "torch",
    "sentencepiece",
    "scipy",
]
```
`"transformers"` lib, in its turn, has a lot of additional requirements. So nncf requirements do not look like a burden in this case.

Reply:
In total nncf installs 33 packages, transformers 16. Unlike transformers, NNCF has things like matplotlib that someone who just wants to use transformer models will never use. Every additional package risks causing issues. We see this with some regularity with openvino-dev dependencies that get conflicts with eachother. If at some point matplotlib decides to only support numpy version X and transformers requires numpy version Y it will be impossible to install optimum-intel[openvino]. It is nice if the base package is as small as possible.

Reply:
BTW, I counted 19 dependencies including torch in the setup.py. But I agree that this can be too much.

Reply:
I added nncf as an additional extra as proposed by @helena-intel